### DISCUSSION TITLE
Unable to deploy ceph-cluster using default config with helm

### DISCUSSION THREAD

**Discussion by caibirdme** [rook/rook#10040] [General]
I deploy a self-hosted k8s cluster and want to deploy a ceph-cluster on that by helm. I just try to run helm install command like: `helm install --create-namespace --namespace rook-ceph rook-ceph-cluster    --set operatorNamespace=rook-ceph rook-release/rook-ceph-cluster`, but it reports error:

```
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: [unable to recognize "": no matches for kind "CephBlockPool" in version "ceph.rook.io/v1", unable to recognize "": no matches for kind "CephCluster" in version "ceph.rook.io/v1", unable to recognize "": no matches for kind "CephFilesystem" in version "ceph.rook.io/v1", unable to recognize "": no matches for kind "CephObjectStore" in version "ceph.rook.io/v1"]
```

Am I missing something?

**Is this a bug report or feature request?**
* Bug Report

**Expected behavior:** a basic available ceph-cluster

**How to reproduce it (minimal and precise):**

`helm install --create-namespace --namespace rook-ceph rook-ceph-cluster    --set operatorNamespace=rook-ceph rook-release/rook-ceph-cluster`

**Environment**:
* OS (e.g. from /etc/os-release): centos
* Kubernetes version (use `kubectl version`): 1.22.0
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): self-hosted

**Comment by travisn** [comment]
@caibirdme Did you install the [operator helm chart](https://rook.github.io/docs/rook/v1.9/helm-operator.html) first? It will install the CRDs for you, looks like they are missing.

**Reply by NadamHL** [reply]
[root@h0082077 tmp]# kubectl get crd cephclusters.ceph.rook.io
NAME                        CREATED AT
cephclusters.ceph.rook.io   2022-09-09T01:39:06Z
[root@h0082077 tmp]# kubectl apply -f cluster.yaml -n rook-ceph
error: unable to recognize "cluster.yaml": no matches for kind "CephCluster" in version "ceph.rook.io/v1"

**Reply by NadamHL** [reply]
kernel: 4.4.198,
k8s: 1.15.12
rook: 1.7.11

**Reply by NadamHL** [reply]
# kubectl get crd cephclusters.ceph.rook.io -o yaml
apiVersion: apiextensions.k8s.io/v1beta1
kind: CustomResourceDefinition
metadata:
  creationTimestamp: "2022-09-09T01:39:06Z"
  generation: 5
  name: cephclusters.ceph.rook.io
  resourceVersion: "170679919"
  selfLink: /apis/apiextensions.k8s.io/v1beta1/customresourcedefinitions/cephclusters.ceph.rook.io
  uid: 48fc1bc5-5e8c-4dd3-a6d5-3af3c4be1a66
spec:
  additionalPrinterColumns:
  - JSONPath: .spec.dataDirHostPath
    description: Directory used on the K8s nodes
    name: DataDirHostPath
    type: string
  - JSONPath: .spec.mon.count
    description: Number of MONs
    name: MonCount
    type: string
  - JSONPath: .metadata.creationTimestamp
    name: Age
    type: date
  - JSONPath: .status.phase
    description: Phase
    name: Phase
    type: string
  - JSONPath: .status.message
    description: Message
    name: Message
    type: string
  - JSONPath: .status.ceph.health
    description: Ceph Health
    name: Health
    type: string
  conversion:
    strategy: None
  group: ceph.rook.io
  names:
    kind: CephCluster
    listKind: CephClusterList
    plural: cephclusters
    singular: cephcluster
  preserveUnknownFields: true
  scope: Namespaced
  subresources:
    status: {}
  validation:
    openAPIV3Schema:
      properties:
        spec:
          properties:
            annotations:
              type: object
            cephVersion:
              properties:
                allowUnsupported:
                  type: boolean
                image:
                  type: string
              type: object
            cleanupPolicy:
              properties:
                confirmation:
                  pattern: ^$|^yes-really-destroy-data$
                  type: string
                sanitizeDisks:
                  properties:
                    dataSource:
                      pattern: ^(zero|random)$
                      type: string
                    iteration:
                      format: int32
                      type: integer
                    method:
                      pattern: ^(complete|quick)$
                      type: string
                  type: object
              type: object
            continueUpgradeAfterChecksEvenIfNotHealthy:
              type: boolean
            dashboard:
              properties:
                enabled:
                  type: boolean
                port:
                  maximum: 65535
                  minimum: 0
                  type: integer
                ssl:
                  type: boolean
                urlPrefix:
                  type: string
              type: object
            dataDirHostPath:
              pattern: ^/(\S+)
              type: string
            disruptionManagement:
              properties:
                machineDisruptionBudgetNamespace:
                  type: string
                manageMachineDisruptionBudgets:
                  type: boolean
                managePodBudgets:
                  type: boolean
                osdMaintenanceTimeout:
                  type: integer
                pgHealthCheckTimeout:
                  type: integer
              type: object
            external:
              properties:
                enable:
                  type: boolean
              type: object
            healthCheck:
              type: object
            logCollector:
              type: object
            mgr:
              properties:
                count:
                  maximum: 2
                  minimum: 0
                  type: integer
                modules:
                  items:
                    properties:
                      enabled:
                        type: boolean
                      name:
                        type: string
                    type: object
                  type: object
              type: object
            mon:
              properties:
                allowMultiplePerNode:
                  type: boolean
                count:
                  maximum: 9
                  minimum: 0
                  type: integer
                volumeClaimTemplate:
                  type: object
              type: object
            monitoring:
              properties:
                enabled:
                  type: boolean
                externalMgrEndpoints:
                  items:
                    properties:
                      ip:
                        type: string
                    type: object
                  type: array
                rulesNamespace:
                  type: string
              type: object
            network:
              properties:
                hostNetwork:
                  type: boolean
                provider:
                  type: string
                selectors:
                  type: object
              type: object
            placement:
              type: object
            removeOSDsIfOutAndSafeToRemove:
              type: boolean
            resources:
              type: object
            security:
              type: object
            skipUpgradeChecks:
              type: boolean
            storage:
              properties:
                config:
                  type: object
                deviceFilter:
                  type: string
                devicePathFilter:
                  type: string
                disruptionManagement:
                  properties:
                    machineDisruptionBudgetNamespace:
                      type: string
                    manageMachineDisruptionBudgets:
                      type: boolean
                    managePodBudgets:
                      type: boolean
                    osdMaintenanceTimeout:
                      type: integer
                    pgHealthCheckTimeout:
                      type: integer
                  type: object
                nodes:
                  items:
                    properties:
                      config:
                        properties:
                          databaseSizeMB:
                            type: string
                          encryptedDevice:
                            pattern: ^(true|false)$
                            type: string
                          journalSizeMB:
                            type: string
                          metadataDevice:
                            type: string
                          osdsPerDevice:
                            type: string
                          storeType:
                            pattern: ^(bluestore)$
                            type: string
                          walSizeMB:
                            type: string
                        type: object
                      deviceFilter:
                        type: string
                      devicePathFilter:
                        type: string
                      devices:
                        items:
                          properties:
                            config:
                              type: object
                            name:
                              type: string
                          type: object
                        type: array
                      name:
                        type: string
                      resources:
                        type: object
                      useAllDevices:
                        type: boolean
                    type: object
                  type: object
                storageClassDeviceSets:
                  type: object
                useAllDevices:
                  type: boolean
                useAllNodes:
                  type: boolean
              type: object
            waitTimeoutForHealthyOSDInMinutes:
              type: integer
          type: object
      type: object
  version: v1
  versions:
  - name: v1
    served: true
    storage: true
status:
  acceptedNames:
    kind: CephCluster
    listKind: CephClusterList
    plural: cephclusters
    singular: cephcluster
  conditions:
  - lastTransitionTime: "2022-09-09T01:39:06Z"
    message: no conflicts found
    reason: NoConflicts
    status: "True"
    type: NamesAccepted
  - lastTransitionTime: null
    message: the initial names have been accepted
    reason: InitialNamesAccepted
    status: "True"
    type: Established
  storedVersions:
  - v1

**Reply by NadamHL** [reply]
> @caibirdme Did you install the [operator helm chart](https://rook.github.io/docs/rook/v1.9/helm-operator.html) first? It will install the CRDs for you, looks like they are missing.

I have made sure that CRD is installed, but there is still an error when installing Cluster. yaml

**Reply by travisn** [reply]
@NadamHL On K8s 1.15 you have `apiextensions.k8s.io/v1beta1` instead of `apiextensions.k8s.io/v1`, so the types are not being found. What version of Rook are you trying to install? Our latest supported release requires at least [K8s 1.17](https://rook.github.io/docs/rook/v1.9/Getting-Started/Prerequisites/prerequisites/#minimum-version).
